### PR TITLE
Memcached

### PR DESCRIPTION
**Pull request by tbwrigh**

### CODE REVIEW

**Comment by tbwrigh** [comment]
@OxygenCobalt can you push new location updates? I don't know how to send a byte body in postman.

**Comment by OxygenCobalt** [comment]
@tbwrigh Here's a script

```py
import struct
from datetime import datetime, timezone
from time import sleep
import requests

coords = [
    (39.7518902, -105.2209817),
    (39.7519935, -105.2208025),
    (39.7521685, -105.2204875),
    (39.7523451, -105.2201381),
    (39.7525108, -105.2198471),
    (39.7526269, -105.2196366),
]

while True:
    for lat, lon in coords[:-1]:
        print(lat, lon)
        timestamp = int(datetime.now(timezone.utc).timestamp())  # Current time in seconds since the Epoch as a float
        # Pack the values into a bytes object
        body = struct.pack("!ldd", timestamp, lat, lon)


        url = "http://localhost:8000/vans/location/0"
        headers = {"Content-Type": "application/octet-stream"}

        response = requests.post(url, data=body, headers=headers)

        print(response.status_code)

        sleep(2)
    for lat, lon in coords[1:][::-1]:
        print(lat, lon)
        timestamp = int(datetime.now(timezone.utc).timestamp())  # Current time in seconds since the Epoch as a float
        # Pack the values into a bytes object
        body = struct.pack("!ldd", timestamp, lat, lon)


        url = "http://localhost:8000/vans/location/0"
        headers = {"Content-Type": "application/octet-stream"}

        response = requests.post(url, data=body, headers=headers)

        print(response.status_code)

        sleep(2)
```

File: backend/src/handlers/vans.py [hunk 1]
```diff
@@ -143,31 +143,40 @@ async def generator():
...
-        raise HTTPException(detail="Van not found", status_code=404)
-
-    location = req.app.state.van_locations[van_id]
-    location_json = {
-        "timestamp": int(location.timestamp.timestamp()),
-        "latitude": location.latitude,
-        "longitude": location.longitude,
-    }
+    # if van_id not in req.app.state.van_locations:
+    #     raise HTTPException(detail="Van not found", status_code=404)
```
Comment (OxygenCobalt) [inline backend/src/handlers/vans.py:147]: Why is this commented?

File: backend/src/handlers/vans.py [hunk 2]
```diff
@@ -179,6 +188,10 @@ async def generator():
 async def post_location(req: Request, van_id: int) -> HardwareOKResponse:
     # byte body: long for timestamp, double for lat, double for lon
     body = await req.body()
+
+    print(body)
+    print(len(body))
```
Comment (OxygenCobalt) [inline backend/src/handlers/vans.py:193]: Left some printf debugging in it seems

File: backend/src/main.py [hunk 3]
```diff
@@ -21,4 +21,4 @@
 @app.on_event("startup")
 def startup_event():
     app.state.db = DBWrapper()
-    app.state.van_locations = TTLCache(maxsize=100, ttl=300)
+    app.state.memcache_client = memcache.Client([("127.0.0.1", 11211)])
```
Comment (OxygenCobalt) [inline backend/src/main.py:24]: Are you sure we want to hard-code these?

**Review by OxygenCobalt — CHANGES_REQUESTED** [review]

**Comment by tbwrigh** [comment]
Decided not to use memcached.

**Pull request closed without merging**